services/dns/wait/wait.go

```diff
@@ -46,8 +46,8 @@ func CreateZoneWaitHandler(ctx context.Context, a APIClientInterface, projectId,
 	return handler
 }
 
-// UpdateZoneWaitHandler will wait for zone update
-func UpdateZoneWaitHandler(ctx context.Context, a APIClientInterface, projectId, instanceId string) *wait.AsyncActionHandler[dns.ZoneResponse] {
+// PartialUpdateZoneWaitHandler will wait for zone update
```

Comment:
Is is relevant for the waiter if the update was partial or not? If there is is both a `PartialUpdate...` and `Update...`, we would only need one waiter for both updates, right?

Reply:
It's so it matches the corresponding method.

Added it to the changelogs and created a `PartialUpdateInstanceWaitHandler` for mongodbflex (which has both `Update` and `PartialUpdate`), which I think is the only case where we have waiters for update operations that can be both partial and full. Please check